### PR TITLE
Delete libuv-iomgr implementation and GRPC_UV build option

### PR DESCRIPTION
**Pull request by drfloob**
This has been unmaintained for years, last supported in gRPC-core v1.24.

### CODE REVIEW

File: test/core/surface/server_chttp2_test.cc [hunk 1]
```diff
@@ -40,7 +40,6 @@ TEST(ServerChttp2, UnparseableTarget) {
 }
 
 // GRPC_ARG_ALLOW_REUSEPORT isn't supported for custom servers
```
Comment (markdroth) [inline test/core/surface/server_chttp2_test.cc:42]: I think this comment can go away too.

File: test/core/surface/server_test.cc [hunk 2]
```diff
@@ -73,7 +73,6 @@ void test_request_call_on_no_server_cq(void) {
 }
 
 // GRPC_ARG_ALLOW_REUSEPORT isn't supported for custom servers
```
Comment (markdroth) [inline test/core/surface/server_test.cc:75]: I think this comment can go away.

**Review by markdroth — APPROVED** [review]
This looks great!  Thanks for doing this!

File: test/core/surface/server_chttp2_test.cc (hunk 1 above)
Comment (drfloob) [reply test/core/surface/server_chttp2_test.cc]: Good catch, done.

File: test/core/surface/server_test.cc (hunk 2 above)
Comment (drfloob) [reply test/core/surface/server_test.cc]: Done.

**Review by murgatroid99 — APPROVED** [review]

**Pull request merged**